perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -206,14 +213,49 @@ public static String join(Iterable<?> collection, String separator) {
 							throw new ConsistencyErrorException("Creating existing extSource: " + additionalExtSourceName);
 						}
 					}
+					UserExtSource userExtSource = new UserExtSource(additionalExtSource, additionalExtLoa, additionalExtLogin);
+
+					List<Attribute> attributes = new ArrayList<>();
+					// Set ues attribute
+					if (additionalExtLoginAndAttribute.length > 1) {
+						List<String> uesAttributesWithValues =  new LinkedList<>();
+						uesAttributesWithValues.addAll(Arrays.asList(additionalExtLoginAndAttribute));
+						uesAttributesWithValues.remove(0);
+						for (String uesAttributeWithValue : uesAttributesWithValues) {
+							String[] uesAttribute = uesAttributeWithValue.split("=");
+							try {
+								Attribute attribute = new Attribute(sess.getPerun().getAttributesManager().getAttributeDefinition(sess, uesAttribute[0]), getAttributeValueFromString(uesAttribute[1]));
+								attributes.add(attribute);
+							} catch (AttributeNotExistsException e) {
+								log.error("User with login {} has invalid attribute for userExtSource defined as {} with value {}.", login, uesAttribute[0], uesAttribute[1]);
+							}
+						}
+					}
+
+					RichUserExtSource richUserExtSource = new RichUserExtSource(userExtSource, attributes);
 					// Add additional user extSource
-					additionalUserExtSources.add(new UserExtSource(additionalExtSource, additionalExtLoa, additionalExtLogin));
+					additionalUserExtSources.add(richUserExtSource);
 				}
 			}
 		}
 		return additionalUserExtSources;
 	}
 
+	/**
+	 * Splits attribute value by ";" and returns list of string values or single string if no ";" occurres.
+	 *
+	 * @param attributeValue unsplitted attribute value
+	 * @return splitted attribute value to list of string or single string
+	 */
+	private static Object getAttributeValueFromString(String attributeValue) {
```

Comment:
The name of this method is very similar to the one parsing attribute value from escaped string (from DB) to the object Attribute in Perun. I would change the name and add more info to the javadoc (because this method has just one purpose and can't be used for other).

Reply:
Changed.